File: lambdas/write-items/handler.js
const { DynamoDBClient } = require("@aws-sdk/client-dynamodb");
const {
  DynamoDBDocumentClient,
  PutCommand,
  DeleteCommand,
} = require("@aws-sdk/lib-dynamodb");
const { randomUUID } = require("crypto");

const clientConfig = {};
if (process.env.AWS_ENDPOINT_URL) {
  clientConfig.endpoint = process.env.AWS_ENDPOINT_URL;
}
const client = new DynamoDBClient(clientConfig);
const docClient = DynamoDBDocumentClient.from(client);
const TABLE_NAME = process.env.TABLE_NAME || "Items";

const response = (status, body) => ({
  statusCode: status,
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify(body),
});

exports.createItem = async (event) => {
  const body = JSON.parse(event.body || "{}");
  const item = {
    ...body,
    name: body.name || "Unnamed",
    createdAt: new Date().toISOString(),
    id: randomUUID(),
  };

  // item.id = randomUUID();

  await docClient.send(
    new PutCommand({
      TableName: TABLE_NAME,
      Item: item,
    }),
  );

  return response(201, item);
};

exports.deleteItem = async (event) => {
  const itemId = event.pathParameters.id;

  await docClient.send(
    new DeleteCommand({
      TableName: TABLE_NAME,
      Key: { id: itemId },
    }),
  );

  return response(200, { message: `Item ${itemId} deleted` });
};

File: lambdas/read-items/handler.py
import json
import boto3
import os

dynamodb = boto3.resource(
    "dynamodb",
    endpoint_url=os.environ.get("AWS_ENDPOINT_URL"),
)
table = dynamodb.Table(os.environ.get("TABLE_NAME", "Items"))


def response(status, body):
    return {
        "statusCode": status,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps(body),
    }


def list_items(event, context):
    """GET /items — return all items."""
    result = table.scan()
    return response(200, result.get("Items", []))


def get_item(event, context):
    """GET /items/{id} — return a single item."""
    item_id = event["pathParameters"]["id"]
    result = table.get_item(Key={"id": item_id})
    item = result.get("Item")
    if not item:
        return response(404, {"error": "Item not found"})
    return response(200, item)
